mypy/checker.py

```diff
@@ -6961,6 +6965,21 @@ def is_literal_not_implemented(n: Expression) -> bool:
     return isinstance(n, NameExpr) and n.fullname == "builtins.NotImplemented"
 
 
+def _is_empty_generator(func: FuncItem) -> bool:
+    """
+    Checks whether a function's body is 'return; yield' (the yield being added only
+    to promote the function into a generator).
+    """
+    return (
+        len(body := func.body.body) == 2
```

Comment:
Nit: I think the first walrus here is slightly gratuitous

```suggestion
    body = func.body.body
    return (
        len(body) == 2
```

Reply:
ebc1cfa193ca64f686587ea91ddb2570f043937a